src/app/core/components/viewchartarea/viewchartarea.component.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  Component, Input, OnDestroy, OnChanges, SimpleChanges, ViewChild, ElementRef,
+} from '@angular/core';
+import { Chart, ChartData } from 'chart.js';
+
+@Component({
+  selector: 'viewchartarea',
+  templateUrl: './viewchartarea.component.html',
+  styleUrls: ['./viewchartarea.component.scss'],
+})
+export class ViewChartAreaComponent implements OnDestroy, OnChanges {
+  @ViewChild('canvas', { static: true }) canvas: ElementRef;
+  @Input() data: ChartData;
+
+  chart: Chart;
+  maxSources = 8;
+
+  makeConfig(data: ChartData): Chart.ChartConfiguration {
+    return {
+      type: 'line',
+      data,
+      options: {
+        responsive: true,
+      },
+    };
+  }
+
+  render(data: ChartData): Chart {
+    if (data.datasets.length > this.maxSources) throw 'ERROR: Maximum Sources Exceeded. Line/Area charts have a hard limit of 8 data sources';
```

Comment:
Please wrap in `new Error()` for reasons described here: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-throw-literal.md
Will add linter rule for this one.